.github/workflows/unit-tests.yml

```
@@ -0,0 +1,36 @@
+name: Unit Tests
+
+on: [pull_request]
+
+jobs:
+  matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      latest: ${{ steps.set-matrix.outputs.requireds }}
+    steps:
+      - uses: ljharb/actions/node/matrix@main
+        id: set-matrix
+        with:
+          versionsAsRoot: true
+          type: majors
+          preset: '>=12'
+
+  test:
+    needs: [matrix]
+    name: 'On Node ${{ matrix.node-version }}'
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: ${{ fromJson(needs.matrix.outputs.latest) }}
+        exclude:
+          - node-version: 13
+          - node-version: 15
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Install Dependencies
+        uses: bahmutov/npm-install@v1
```

this seems kind of weird, rather than just:
```suggestion
      - run: yarn install
```

---

This `bahmutov/npm-install@v1` action handles `node_module` caching.

---

ah, interesting. not sure that's really beneficial on a package's CI, but if it works for you, great